statsmodels/tsa/arima/params.py

```diff
@@ -351,6 +575,86 @@ def is_invertible(self):
 
         return ma_stationary and seasonal_ma_stationary
 
+    def set_fixed_params(self, fixed_params, validate=False):
+        """
+        Sets all fixed params at once, which can be accessed afterwards through
+        `fixed_exog_params`, `fixed_ar_params`, etc.
+
+        Parameters
+        ----------
+        fixed_params : dict
+            Dictionary with names of fixed parameters as keys (e.g. 'ar.L1',
+            'ma.L2'), which correspond to SARIMAXSpecification.param_names.
+            Dictionary values are the values of the associated fixed
+            parameters.
+        validate : bool, optional
+            Whether to validate fixed parameter names and values.
+            See `SARIMAXSpecification.split_fixed_params` function.
+        """
+
+        split_fixed_params, split_is_fixed_param = (
+            self.spec.split_fixed_params(
+                fixed_params=fixed_params, validate=validate
+            )
+        )
+        param_types = split_fixed_params.keys()
+        assert set(param_types) == set(self._params_split) - {"sigma2"}
+
+        for param_type in param_types:
+            # write to param_split
+            self._set_param_array(
+                param_type=param_type, value=split_fixed_params[param_type],
+                k_params=split_is_fixed_param[param_type].sum(),
+                title=None, bool_indexer=split_is_fixed_param[param_type]
+            )
+            # update is_param_fixed_split
+            self._is_param_fixed_split[param_type] = (
```

Comment:
Allows overwriting already fixed parameters to be consistent with other existing implementations, in which case a warning would be raised (see `_warn_fixed_overwrite` below).

Reply:
This is great. And if I understand correctly, the warning does not appear when you overwrite previously fixed parameters with *new* fixed parameters using `set_fixed_params`, but it does appear when you try to set the values of free parameters over the top of previously fixed parameters, e.g. using the `ar_params` setter. Is that right?